data/microos/butane/script

```diff
@@ -79,28 +82,64 @@ test_prepare() {
     echo "combustion: test_prepare function ran OK" > /dev/kmsg
 }
 
+can_encrypt() {
+	if [ -c "/dev/tpm0" ] && command -v disk-encryption-tool >/dev/null; then
+		return 0
+	else
+		return 1
+	fi
+}
+
 if [ "${1-}" = "--prepare" ]; then
     test_prepare
+
+	if can_encrypt; then
+        mkdir -p /run/credstore
+        echo "force" > /run/credstore/disk-encryption-tool-dracut.encrypt
+        echo "cr_root /dev/vda3" >> /run/credstore/disk-encryption-tool-dracut.partitions
+    fi
+
     exit 0
 fi
 
 # Redirect output to the console
 exec > >(exec tee -a /dev/console) 2>&1
 
+if ! grep -q tumbleweed /etc/os-release; then
```

Comment:
Another nitpick:

I believe that instead of creating new processes in an initialization script we could just `source /etc/os-release` and check the `ID` variable.

Reply:
I have sourced the os-release and used the approriate vars